package/com.unity.formats.usd/Runtime/Scripts/IO/SanitizedSamples.cs

```diff
@@ -544,15 +544,24 @@ void ConvertInterpolationToFaceVarying<T>(ref T[] values, int[] vertexIndices, b
             if (values == null)
                 return;
 
-            var newInterpolation = GuessInterpolation(values.Length);
+            // We assume a certain number of values in the array for certain interpolations, so check whether this assumption will work and not cause array overflows
+            bool interpolationIsValid = false;
```

Comment:
can we add a
`if interpolation == null or values == null; return false`
in the IsInterpolationValid function and then do 

```
if (IsInterpolationValid) {
...
}
```

right away?

Reply:
I agree that it would be clearer if the `if interpolation == null` were inside the `IsInterpolationValid`, I've made that change :) 
However I think let's leave the values check outside that function, because 1. it adds an unnecessary function call in the case that values are null which is currently an early out (it may be compiled away, but I'm not sure), and 2. values being null has more significance than 'just' the interpolation being invalid, so logically I prefer it outside that function. What do you think of this change?

Reply:
That sounds good! Thank you for the change!